operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -687,17 +688,6 @@ func (rc *ReconciliationContext) CheckRackScale() result.ReconcileResult {
 				return result.Error(err)
 			}
 		}
-
-		currentReplicas := statefulSet.Status.CurrentReplicas
-		if currentReplicas > desiredNodeCount {
```

Comment:
Would it be better to leave this conditional with an additional clause to check if we are not scaling down?

Reply:
So with the current logic, we don't attempt a scale down until the end of the reconcile loop after everything is happy. At this point in the code, we wouldn't have that condition set yet :(